### DISCUSSION TITLE
Controlling the Link status LED

### DISCUSSION THREAD

**Discussion by Android-X13** [arendst/Tasmota#16790] [Q&A] [answered]
I have a Shelly Plug S activated with the recommended [template](https://templates.blakadder.com/shelly_plug_S.html).
The plug has two LEDs, a red power status light and a blue link status light.

I am able to control the red LED with commands like `LedPower` and switch it on/off, make it blink, etc, but I don't know how to control the blue LED. The only way for me to have access to it is through the `LedState` commands, but the `LedPower<x>` commands cannot be used with it.

Is it possible to control that LED?

**Comment by sfromis** [comment, marked as answer]
If you want to take full control, config the pin as a Relay or PWM, both giving you power on/off, the latter additionally giving you brightness control.

**Reply by Android-X13** [reply]
Configuring it as `PWM_i` indeed lets me control the LED with `Power2` command however I'm having other undesirable effects: `LedState 1` does not work (the other LED about power status does not change when power state changes)

Any clue on how to fix this?

**Reply by sfromis** [reply]
With only one Led-configure pin left, that one is shared to get a link status led still. You have other commands to influence usage of this.
https://tasmota.github.io/docs/Lights/#status-leds
https://tasmota.github.io/docs/Commands/#ledpower

**Reply by Android-X13** [reply]
I already know about the `LedPower` command.
I'm not sure I understand, do you mean that the power status LED is now being used as both a power and a link status LED, and this behavior cannot be changed?
I also know that I can get rid of the link status led with `LedState 0` but this isn't what I'm after

**Reply by sfromis** [reply]
Yes, no isolated power status led, as a link status is seen as important.
> With only one Led-configure pin left, that one is shared to get a link status led still.

As I do not worry much about the plug leds, I've not worked out details of how the combinations of options work there, and cannot say exactly what you can or cannot get.

**Reply by barbudor** [reply]
If you want your red led not to be influenced by link status you must define a GPIO as LedLink. It doesn't need to be a GPIO where a led or anything is actually connected.
However with 1 relay, 1 pwm(the 2nd led), one Led_1 and a dummy LedLink, you might face other issues as Led_1 should reflect the power status of both Relay and PWM

**Reply by Android-X13** [reply]
@barbudor I'd like to test your suggestion, do I set (as LedLink) a GPIO which is currently set to "None" or a GPIO that is set to "User"?

**Reply by barbudor** [reply]
If possible, I would first choose a User one if you don't need it
A None one is OK too if it is not on the red list in which case some care has to be taken

**Reply by Android-X13** [reply]
In the template I linked above there are 2 "None" (colored in red) and 4 "User" GPIOs. I don't know whether I need them or not, I don't even know what their difference is... I guess I'll randomly choose one of the "User" GPIOs

**Reply by barbudor** [reply]
Generally in a template those words means:
- User is a GPIO which can be re-used for DIY, either because it is made available through a connector reachable externally (eg Shelly 1) or internatlly (eg Sonoff Basic), or at least easily reachable with a bit of soldering skills.
- None is supposed to be a GPIO that can't be used either because it is illegal (like 9 and 10) or because it is not reachable (like there is no connector or easy way to reach it without very good soldering skills)

In the specific case of the Shelly Plug S which is a power monitoring device, DIY is strongly discouraged as the whole circuit is not isolated from the AC because of the power measurment circuit.

**Reply by sfromis** [reply]
For the special case of wanting to use a gpio as a "dummy" output without wanting to have any connections, in most cases it would be ok to use a non-reachable "None" pin, as long as it is not the red-marked gpio 9 or 10. Thus, choosing any of the "User" should be fine.

The function of designating a pin as "User" in the template is that it becomes available in Configure Module.

**Reply by Android-X13** [reply]
This is what happens:

Configuring GPIO2 as `PWM_i` (previously `LedLink_i`):
- Allows control of the LED with `Power2` command
- Power status LED (red) does not change when power state changes (`LedState 1` does not work)
- Power status LED (red) is blinking when there's no access to the WiFi (instead of the blue link status LED)

Configuring GPIO2 as `PWM_i` (previously `LedLink_i`) and GPIO1 as `LedLink` (previously `User`):
- Allows control of the blue LED with `Power2` command
- Power status LED (red) changes when power state changes (`LedState 1` works)
- No led is blinking when there's no access to the WiFi

Ideally what I'd like to happen is:
- Control the blue LED with `Power2` command
- Power status LED (red) changing when power state changes (`LedState 1` working, as per default)
- Blue LED blinking when there's no access to the WiFi (as per default)

It's not that big of a deal though and I can live with it, now that I can control both leds. Thank you both for your help.

PS. In 2nd case I didn't notice any issues with Led_1 (red) reflecting the power status of both Relay and PWM. It only reflects status of the Relay

**Reply by barbudor** [reply]
I think you didn't get the explanations

You can't have power 2 status and link status together
Either link status is mixed with the Power1 status, either it is on its own. That's all
